src/components/NewAccountPage/NewAccountPage.jsx

```diff
@@ -150,36 +196,29 @@ export default () => {
                <StyledLabel>Password</StyledLabel>
                <StyledField
                   required
-                  type="text"
-                  placeholder="Password"
+                  type={passwordShown ? 'text' : 'password'}
                   defaultValue={password}
                   onChange={(e) => setPassword(e.target.value)}
                />
+               <StyledHideButton onClick={toggleShowPassword}>
+                  {passwordShown ? 'Hide ' : 'Show '}
+                  {passwordShown ? (
+                     <FontAwesomeIcon icon={faEye} />
+                  ) : (
+                     <FontAwesomeIcon icon={faEyeSlash} />
```

Comment:
I don't think you need two different icons, just use the same icon but on click it toggles show password. Also did you figure out how to align it properly on the right?